tools/toolchains/mbed_toolchain.py

```diff
@@ -1099,23 +1099,33 @@ def check_and_add_minimal_printf(self, target):
         ):
             self.flags["common"].append("-DMBED_MINIMAL_PRINTF")
 
-    def check_c_lib_supported(self, target, toolchain):
-        """
-        Check and raise an exception if the requested C library is not supported,
+    def get_c_lib_config(self, target, toolchain):
+        """get the c library configuration from the target if not returns std by default."""
+        c_lib = "std"
+        if hasattr(target, "c_lib"):
+            c_lib = target.c_lib.lower()
+        elif (
+              hasattr(target, "selected_c_libs")
+              and toolchain in target.selected_c_libs
+        ):
+            c_lib = target.selected_c_libs[toolchain];
+        elif (
+              hasattr(target, "default_build")
+              and toolchain == "gcc_arm"
+        ):
+            c_lib = target.default_build
```

Comment:
Why not `target.selected_c_libs[toolchain].lower()` and `target.default_build.lower()`?